bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/clangd/ClangdCProjectDescriptionListener.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Bachmann electronic GmbH and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Gesa Hentschke (Bachmann electronic GmbH) - initial implementation
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp.clangd;
+
+import org.eclipse.cdt.core.settings.model.CProjectDescriptionEvent;
+import org.eclipse.cdt.core.settings.model.ICProjectDescription;
+import org.eclipse.core.resources.IProject;
+
+/**
+ * Vendors can implement this interface as OSGi service
+ * with a service.ranking property > 0 to implement custom behavior
+ * and to replace the {@link ClangdConfigurationManager}
+ */
+public interface ClangdCProjectDescriptionListener {
```

Comment:
this interface needs more work, it has 3 methods with tightly-coupled implementation and all of them are open for public invocation. Ideally there should be only one.

Reply:
Should be fixed with my last commit